### ISSUE TITLE
[BBCode/Markdown] Shrink/expand long code blocks

### ISSUE THREAD

**Issue by jordanjay29** [flarum/core#683]
Example: https://discuss.flarum.org/d/1658-401-unauthorized/22

There really needs to be a way to shrink/expand long code blocks in a forum post so that they don't obnoxiously fill the screen. Scrollbars would be one way to do it, or have a javascript (JQuery?) shrink/expand function which would be more "modern" and definitely more mobile friendly.

**Comment by franzliedke** [comment]
Very good idea, I agree. I kind of like the "expand" idea, that in combination with scrollbars might be best.

**Comment by matpompili** [comment]
I'm working on it. I think I'll use this script: https://github.com/jedfoster/Readmore.js 
What do you think?

**Comment by jordanjay29** [comment]
Can it be localized? I think the encapsulation of the language strings might be a problem.

**Comment by matpompili** [comment]
Yes it can be localized. In the Use section of the README.md there are lots of options to customize the script, such as more and less strings

**Comment by JoshyPHP** [comment]
Personally I'd rather see some sort of thin div at the bottom with a `⌄` or `⌃` symbol that either expands or shrinks the code block when clicked.

**Comment by tobscure** [comment]
I agree with @JoshyPHP. Also, for the sake of file size, unless we can find a microlibrary that does exactly what we need in only a few lines, we can probably just write our own JS - e.g. https://github.com/esotalk/esoTalk/blob/develop/core/js/conversation.js#L581

**Comment by tobscure** [comment]
@franzliedke and I discussed this and agree that we should implement an "expand" toggle button which increases the height of a code block/quote so that all of its contents is visible.
